android/runtime/common/src/js/module.js

```diff
@@ -571,20 +569,25 @@ Module.prototype._runScript = function (source, filename) {
 				return inspectorWrapper(source, filename);
 			}
 		}
-		// run app.js "normally" (i.e. not under debugger/inspector)
-		return Script.runInThisContext(source, filename, true);
+		// continue to run app.js "normally" (i.e. not under debugger/inspector)
 	}
 
 	const ti = new Titanium.Wrapper(context);
 
-	// In V8, we treat external modules the same as native modules.  First, we wrap the
-	// module code and then run it in the current context.  This will allow external modules to
-	// access globals as mentioned in TIMOB-11752. This will also help resolve startup slowness that
-	// occurs as a result of creating a new context during startup in TIMOB-12286.
-	source = Module.wrap(source);
-
-	const f = Script.runInThisContext(source, filename, true);
```

Comment:
This is the underlying reason why the `setTimeout` is not defined Errors are happening, AFAIK. Basically before we ran in "this" context, i.e. we passed along our main app global context and that's what the "module"/script ran under. Basically it shared the current context, so changes were reflected in the current (global) context.

With your new code there's no way to run in "this" context, so everything is isolated. In both codepaths (runAsModule, runAsScript) you're always creating a new context under which the module executes. So while we can now pass in a context object as the 3rd arg, basically it's a one-way copy from this context to the created one under the hood - i.e. we can tell it to pass along require/Ti/global, etc to be set on the global/context of the module, but that doesn't get "copied" back. Note also that just copying a reference to the global likely won't behave as expected, since contexts define new globals (see more below).

Now, I don't know if we need to retain some way to specify to run in "this" context still to fix this, or maybe when you generate your new context you can pass in the pre-existing global object to it (in `Context::New`) - I don't know if changes made to the global in that new context will be "wiped away" once that new context is "done", or if they'd persist outside the module.

I suspect this may also be why in my own PR around require I was seeing issues in extending Ti.Blob from JS suddenly - it's possible the extensions done in the bootstrapping (`kroll.js`) vs app startup (`ti.main.js`) may affect the global differently due to the context the module/script is executed under.